BITNODE: CHARITY - preview of a wip - DO NOT MERGE

Sphyxis opened the pull request:
This is a current wip.  The charity BN.

Currently working:
Slums offer a new type of task, charity work.
Sleeves can be automated to do Charity work.
City Hall will allow you to start a Charity (Works similar to Gangs) (Not fully implemented yet)
The Lottery store is now in Sector-12
Along with the lottery store comes the ns.lottery command, to help automate all your lottery ticket purchases.

Please let me know what you think!

Sphyxis committed:
BN-Charity - first upload

Added:
Charitable acts in the slums
Sleeves can do charitable acts
Added a charity organization that can be created in Sector-12 City Hall
Added a Lottery Store in Sector-12

Sphyxis committed:
Working Lotto store

We have Lotto 6/49, Keno, Pick2, Pick3, Pick4 to choose from.  Tickets do not save with person yet.

Sphyxis committed:
Lottery ready.  API added

Finished up the lottery house, added rules and win amounts to all games.  Added ns.lottery to automate ticket purchases.

Sphyxis committed:
Updated tests

ficocelliguy commented:
Can you mark this as a draft if it should not be merged?

Also, can you add screenshots, and describe more in detail what optimization problem this bitnode gives the player to solve?

Sphyxis commented:
Optimization problems:
It's simple.  Run your charity, raise money, level it's members and get both it's visibility to 100 and terror to 0.  Keep it there for a few seconds to unlock the end prize of the BN - permanent, unupgradable personal servers that follow you through resets.  If you've done Gangs you'll know how to do this as it's based off of it.  The higher your visibility and the lower your terror the higher your general modifier is.
The higher your visibility though, the higher the drain on it.  Same with Terror, the lower it is the more it will fight back.
You need to manage your charities resources.  It has it's own bank and a cost to actions and equipment.  If you run out of money, actions that require spending will stop.
You will be able to spend your Karma on certain boosts, such as Time Skip, Add bonus time to Sleeves, add bonus time to any area (Such as Charities, Gangs, Bladeburner, etc), adding to facton rep, etc.
Charities will require some positive karma before they can be started, so there are positive actions in the slums that your sleeves can participate in as well.

And, because I was told there should be a little something extra,

I've added a fully functioning lottery system, that can be accessed through the lottery store in Sectore-12 (By the WSE) and full API access.  It's drawn numbers are based off of the Casino's faulty RNG system with a similar max limit imposed on earning.  However, when you break it, it unlocks Ice Cream!

![Cap 6](https://github.com/bitburner-official/bitburner-src/assets/148732459/542bde21-1726-4863-bec0-cb7b724344a1)
![Cap 5](https://github.com/bitburner-official/bitburner-src/assets/148732459/c74f7a68-14b9-48e1-88e5-4519a713d3f7)
![Cap 4](https://github.com/bitburner-official/bitburner-src/assets/148732459/f7b28d0a-8159-46d1-90c1-179dc10ac3ff)
![Cap 3](https://github.com/bitburner-official/bitburner-src/assets/148732459/37194209-5a58-49c9-b816-3e41bbf8d3b4)
![Cap 2](https://github.com/bitburner-official/bitburner-src/assets/148732459/b81cdb3e-50ca-4be8-a7df-88ecaf9bb0fa)
![Cap 1](https://github.com/bitburner-official/bitburner-src/assets/148732459/2981b314-1690-4b35-a3d6-988b5fffa67c)

Sphyxis committed:
Working interface

Sphyxis committed:
Updated API Access, completed general playthough

The API has been added.  It's only missing the API ability to spend Karma on things.
Karma spending page is in progress, but most is done.
Lottery Store should be done.

Sphyxis committed:
just npm format

Needed to re-run npm format
